Nostr_relay/relay/src/server.rs

```diff
@@ -93,14 +111,265 @@ impl Handler<Disconnect> for Server {
+
+        let (sender, receiver) = mpsc::channel();
+
+        // let sender = Arc::new(sender);
+        rt.block_on(async move {
+            let grpc_server_addr = format!("http://{}", zi);
+
+            let mut client = ZchronodClient::connect(grpc_server_addr).await.unwrap();
+
+
+            let request = tonic::Request::new(Empty {});
+
+            match client.query_poll_list(request).await {
+                Ok(response) => {
+                    let pl = response.into_inner().poll_list;
+                    println!("Received response: {:?}", &pl);
+                    sender.send(pl).unwrap();
+                }
+                Err(_) => {
+                    println!("failed to send")
+                }
+            }
+        });
+        let received_vec = receiver.recv().unwrap();
+        drop(rt);
+        received_vec
+    }
+    async fn send_to_chronod_event(&self, e: Event) {
+        println!("send to chronod_event here");
+        let mut rt = Runtime::new().unwrap();
+        //let zi = self.zchronod_ip.clone();
+        let zi = "127.0.0.1:10020".to_string();
+
+        println!("in send to chronod future");
+        let e_c = e.clone();
+        let grpc_server_addr = format!("http://{}", zi);
+
+        let mut client = ZchronodClient::connect(grpc_server_addr).await.unwrap();
+
+        let t: Vec<TagArray> = e_c.tags().into_iter()
+            .map(|inner_vec| TagArray { values: inner_vec.clone() })
+            .collect();
+        let request = tonic::Request::new(ZchronodRequest {
+            msg: Some(c_Event {
+                id: vec![1],
+                pubkey: vec![6, 7],
+                created_at: 2,
+                kind: 2,
+                tags: vec![],
+                content: "hello12333".to_string(),
+                sig: vec![7, 8, 9, 10],
+            })
+        });
+        info!("should send");
+        // let request = tonic::Request::new(ZchronodRequest {
+        //     msg: Some(c_Event {
+        //         id: Vec::from(e_c.id()),
+        //         pubkey: Vec::from(e_c.pubkey()),
+        //         created_at: get_current_system_time(),
+        //         kind: e_c.kind() as u32,
+        //         tags: t,
+        //         content: e_c.content().clone(),
+        //         sig: Vec::from(e_c.sig()),
+        //     })
+        // });
+        let response = client.send(request).await.unwrap().into_inner();
+        println!("Received response: {:?}", response);
+        // match client.send(request).await {
+        //     Ok(response) => {
+        //         println!("Received response: {:?}", response.into_inner());
+        //     }
+        //     Err(_) => {
+        //         println!("failed to send")
+        //     }
+        // }
+
+        //drop(rt);
+    }
+}
+
+async fn query_poll_list(tx: Sender<Vec<String>>, eventid: String) {
+    let mut rt = Runtime::new().unwrap();
+    //let zi = self.zchronod_ip.clone();
+    let zi = "127.0.0.1:10020".to_string();
```

Comment:
the zchronod could be deployed in other nodes, so should apply a new design to et zchronod`ip to adapt to more scenarios. Fine in current scenario.